MapleServer2/PacketHandlers/Login/CharacterManagementHandler.cs

```diff
@@ -90,29 +99,34 @@ public static void HandleCreate(LoginSession session, PacketReader packet)
             Account account = DatabaseManager.Accounts.FindById(session.AccountId);
             Player newCharacter = new Player(account, name, gender, job, skinColor);
 
-            int equipCount = packet.ReadByte();
+            byte equipCount = packet.ReadByte();
             for (int i = 0; i < equipCount; i++)
             {
-                uint id = packet.ReadUInt();
+                int id = packet.ReadInt();
                 string typeStr = packet.ReadUnicodeString();
                 if (!Enum.TryParse(typeStr, out ItemSlot type))
                 {
                     throw new ArgumentException($"Unknown equip type: {typeStr}");
                 }
+
                 EquipColor equipColor = packet.Read<EquipColor>();
 
                 switch (type)
                 {
                     case ItemSlot.HR: // Hair
                         // Hair Length/Position
-                        float backLength = BitConverter.ToSingle(packet.Read(4), 0);
+                        float backLength = packet.ReadFloat();
                         CoordF backPositionCoord = packet.Read<CoordF>();
                         CoordF backPositionRotation = packet.Read<CoordF>();
-                        float frontLength = BitConverter.ToSingle(packet.Read(4), 0);
+                        float frontLength = packet.ReadFloat();
                         CoordF frontPositionCoord = packet.Read<CoordF>();
                         CoordF frontPositionRotation = packet.Read<CoordF>();
+                        if (!DefaultItemsMetadataStorage.IsValid((int) job, id))
```

Comment:
Can't this check just be done after line 105?

Reply:
No because depending on the item slot type, there's more packets to read. If this was done after line 105, it would jumble everything up.

Reply:
Yeah that makes sense, good point